tests/stream_reader_test.py

```diff
@@ -0,0 +1,56 @@
+import time
+
+import numpy as np
+import pytest
+
+from BusinessTampereTrafficMonitoring.stream_reader.stream_reader import StreamReader
+
+# Sample video from https://sample-videos.com
+SOURCE = "tests/samples_for_tests/test.mp4"
+
+
+@pytest.mark.parametrize("source", list("01345678:"))
+def test_init_class_with_invalid_source(source):
+    with pytest.raises(IOError):
+        StreamReader(source)
+
+
+def test_init_class_with_valid_source():
+    sr = StreamReader(SOURCE)
+    assert sr.cap.isOpened() is True
+
+
+def test_get_constants():
+    sr = StreamReader(SOURCE)
+    assert sr.get_constants() == (320, 160)
+
+
+def test_find_nearest_valid_input():
+    test_array = np.array([1, 4, 7, 8, 12, 14, 17, 19, 25, 26])
+    assert StreamReader.find_nearest(test_array, 3) == 4
+    assert StreamReader.find_nearest(test_array, 12.5) == 12
+    assert StreamReader.find_nearest(test_array, 18) == 17
+    assert StreamReader.find_nearest(test_array, 21) == 19
+    assert StreamReader.find_nearest(test_array, 10.5) == 12
+    assert StreamReader.find_nearest(test_array, 25.5) == 25
+    assert StreamReader.find_nearest(test_array, 9999) == 26
+    assert StreamReader.find_nearest(test_array, -123453) == 1
```

Comment:
This one on the other hand could have been simplified with @pytest.parametrize, but this is fine too

Reply:
I will leave it like this for now, I have to prioritize a bit.